### PR TITLE
Docs: Fix js options to what is really inside js

### LINKED ISSUE #36936: Collapse.Options.parent default value: Type 'boolean' is not assignable to type 'string | Element | JQuery'
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

`Collapse.Options.parent` default value is `false`. It is `boolean` type.
`Collapse.Options.parent` type is `string | Element | JQuery` and can't defined as `boolean` type.

`options: Collapse.Options = { parent: false, toggle: true }`

ts error:

> Type 'boolean' is not assignable to type 'string | Element | JQuery'.

### Reduced test cases

from: https://getbootstrap.com/docs/5.2/components/collapse/#options

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.2.0

### PR DESCRIPTION
**Pull request by louismaximepiton**
Fixes #36936.

### CODE REVIEW

File: site/content/docs/5.2/components/collapse.md [hunk 1]
```diff
@@ -146,7 +146,7 @@ const collapseList = [...collapseElementList].map(collapseEl => new bootstrap.Co
 {{< bs-table "table" >}}
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-`parent` | selector, jQuery object, DOM element | `false` | If parent is provided, then all collapsible elements under the specified parent will be closed when this collapsible item is shown. (similar to traditional accordion behavior - this is dependent on the `card` class). The attribute has to be set on the target collapsible area. |
```
Comment (louismaximepiton) [inline site/content/docs/5.2/components/collapse.md:149]: Not sure if I should remove `jQuery object` but it's its only reference in doc.

File: site/content/docs/5.2/components/scrollspy.md [hunk 2]
```diff
@@ -377,9 +377,9 @@ const scrollSpy = new bootstrap.ScrollSpy(document.body, {
 {{< bs-table "table" >}}
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `rootMargin` | string | `0px 0px -40%` | Intersection Observer [rootMargin](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/rootMargin) valid units, when calculating scroll position. |
+| `rootMargin` | string | `0px 0px -25%` | Intersection Observer [rootMargin](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/rootMargin) valid units, when calculating scroll position. |
 | `smoothScroll` | boolean | `false` | Enables smooth scrolling when a user clicks on a link that refers to ScrollSpy observables. |
-| `target` | string \| jQuery object \| DOM element |  | Specifies element to apply Scrollspy plugin. |
+| `target` | DOM element | `null` | Specifies element to apply Scrollspy plugin. |
```
Comment (GeoSot) [inline site/content/docs/5.2/components/scrollspy.md:382]: it resolves strings to elements too (could use `selector`) :wink:

File: site/content/docs/5.2/components/offcanvas.md [hunk 3]
```diff
@@ -302,7 +302,7 @@ const offcanvasList = [...offcanvasElementList].map(offcanvasEl => new bootstrap
 {{< bs-table "table" >}}
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
-| `backdrop` | boolean or the string `static` | `true` | Apply a backdrop on body while offcanvas is open. Alternatively, specify `static` for a backdrop which doesn't close the offcanvas when clicked. |
+| `backdrop` | boolean, `static` | `true` | Apply a backdrop on body while offcanvas is open. Alternatively, specify `static` for a backdrop which doesn't close the offcanvas when clicked. |
```
Comment (GeoSot) [inline site/content/docs/5.2/components/offcanvas.md:305]: I suppose it was ok as it was

File: site/content/docs/5.2/components/collapse.md (hunk 1 above)
Comment (GeoSot) [reply site/content/docs/5.2/components/collapse.md:149]: if it is as you say, you may remove it and in another PR we can use a term like `resolvable element` with the meaning of `selector, jQuery object, DOM element`

**Comment by GeoSot** [comment]
Plus, you may include https://github.com/twbs/bootstrap/issues/36936 too

File: site/content/docs/5.2/components/scrollspy.md (hunk 2 above)
Comment (louismaximepiton) [reply site/content/docs/5.2/components/scrollspy.md:382]: Ah yep, you're right, I just took the js description, maybe we should update the js then in order to receive `string | element` (actually it's only `element` in https://github.com/twbs/bootstrap/blob/main/js/src/scrollspy.js#L51) ?

**Comment by louismaximepiton** [comment]
> Plus, you may include #36936 too

I think this is already taken into account in https://github.com/twbs/bootstrap/pull/37019/files#diff-eea88f447d7813cdc4b41d8d41bfd86111df371dc511a094bf8af223fe07cb00R149. Am I wrong ?

**Comment by GeoSot** [comment]
Sorry, my mistake

File: site/content/docs/5.2/components/scrollspy.md (hunk 2 above)
Comment (GeoSot) [reply site/content/docs/5.2/components/scrollspy.md]: the js has to stay as it was. It is resolved before the typeChecking

Comment (louismaximepiton) [reply site/content/docs/5.2/components/scrollspy.md]: Oh ok, I'll revert those, sorry for my misunderstanding!

**Review by GeoSot — APPROVED** [review]

**Pull request merged**